src/components/MemoList.jsx

```diff
@@ -24,8 +26,8 @@ export default function MemoList({
         <div>
           <Header level={2}>Detail View</Header>
           <textarea value={editedContent} onChange={onEditContentChange} />
-          <Button onClick={onDeleteMemo}>Delete</Button>
-          <Button onClick={onUpdateMemo}>Update</Button>
+          {auth && <Button onClick={onDeleteMemo}>Delete</Button>}
+          {auth && <Button onClick={onUpdateMemo}>Update</Button>}
```

Comment:
[imo]
同じ条件なのでまとめてしまってもいいかもですね〜
もしくは、ボタンを消してしまうのではなく、disabled にするなどした方がユーザはわかりやすいかなーと思ったんですがどうでしょうか！
```suggestion
          <Button onClick={onDeleteMemo} disabled={!auth}>Delete</Button>
          <Button onClick={onUpdateMemo} disabled={!auth}>Update</Button>
```

Reply:
> もしくは、ボタンを消してしまうのではなく、disabled にするなどした方がユーザはわかりやすいかなーと思ったんですがどうでしょうか！

ありがとうございます！
ボタンが表示されないよりも非活性にした方がユーザーにとって分かりやすいと私も思います！

ご教示頂いたように修正したのですが、`disabled={!auth}`が機能せずログアウト時でもボタンが押せるようになってしまいます。
`auth`の状態を`console.log`で確認したのですが、ログイン時は`true`、ログアウト時は`false`が正しく返ってきていました。
サーバーの再起動やリロードなどいろいろ試したのですが解決できなかったため、お手数おかけしますがお力添えいただけますと幸いです🙏💦